pkg/controller/node/node_controller.go

```diff
@@ -424,7 +432,8 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 	// TODO: Deep-copy only when needed.
 	pool := machineconfigpool.DeepCopy()
 	everything := metav1.LabelSelector{}
-	if reflect.DeepEqual(pool.Spec.MachineSelector, &everything) {
+
+	if reflect.DeepEqual(pool.Spec.MachineSelector, &everything) && reflect.DeepEqual(pool.Spec.NodeSelector, &everything) {
 		ctrl.eventRecorder.Eventf(pool, v1.EventTypeWarning, "SelectingAll", "This machineconfigpool is selecting all machines. A non-empty selector is required.")
```

Comment:
can we change `all machines` to `all nodes` here as well?

Reply:
sure!